src/api/custom-resolvers/mutations/create-many-applications/index.ts

```diff
@@ -0,0 +1,101 @@
+import { CreateManyApplicationsVariables, Application, Context, ObjectType } from '../../../types';
+import getCountries from '../../../helpers/get-countries';
+import { mockApplicationCreationObject } from '../../../test-mocks/mock-application';
+
+/**
+ * createManyApplications
+ * Creates many applications.
+ * This is used by E2E tests only
+ * 1) Creates an empty object array with the number of elements passed in variables
+ * 2) Gets all countries and creates a buyer with the first country
+ * 3) Maps application creation object
+ * 4) Creates many applications
+ * 5) Creates reference numbers for applications and updated ReferenceNumber and Application tables
+ * 6) returns applications array and success flag
+ * @param {Object} root: GraphQL root variables
+ * @param {CreateManyApplicationsVariables} GraphQL variables for the createManyApplications mutation
+ * @param {Context} context: KeystoneJS context API
+ * @returns {Promise<Object>} Object with success flag and applications array
+ */
+const createManyApplications = async (root: any, variables: CreateManyApplicationsVariables, context: Context) => {
+  console.info('Creating many applications');
+
+  try {
+    const emptyArray = new Array(variables.count).fill({});
+
+    const countries = await getCountries(context);
+
+    // creates buyer with first country from countries array
+    const buyer = await context.db.Buyer.createOne({
+      data: {
+        country: {
+          connect: {
+            id: countries[0].id,
+          },
+        },
+      },
+    });
+
+    // creates object for creation/update of applications
+    const mockApplicationsData = emptyArray.map(() => ({
+      owner: {
+        connect: {
+          id: variables.accountId,
+        },
+      },
+      buyer: {
+        connect: {
+          id: buyer.id,
+        },
+      },
+      ...mockApplicationCreationObject,
+    }));
+
+    // creates many applications
+    const applications = (await context.db.Application.createMany({
+      data: mockApplicationsData,
+    })) as Array<Application>;
+
+    // generates data for creating referenceNumbers
+    const referenceNumbersData = applications.map((application) => ({
+      application: {
+        connect: {
+          id: application.id,
+        },
+      },
+    }));
+
+    // creates reference numbers
+    const referenceNumbers = await context.db.ReferenceNumber.createMany({ data: referenceNumbersData });
+
+    // maps update object for updating all applications with relevant reference numbers
+    const updateApplicationReferenceNumbers = referenceNumbers.map((referenceNumber: ObjectType) => ({
+      where: { id: referenceNumber.applicationId },
+      data: { referenceNumber: referenceNumber.id },
+    }));
+
+    // updates application's reference numbers
+    await context.db.Application.updateMany({
+      data: updateApplicationReferenceNumbers,
+    });
+
+    const allApplications = await context.db.Application.findMany();
```

Comment:
Might not need to do this if we use the response from `Application.updateMany({ ...`

Reply:
its required, its returned in the wrong order otherwise which breaks tests

Reply:
What do you mean by wrong order? If it's an ascending/descending issue, could use `.reverse()`

Reply:
tried that, its not working.  The only thing that works is getAll again.  I have no idea why it doesn't work

Reply:
What doesn't work? Could try doing this instead 

```ts
context.queryApplication.findMany({
  query: 'referenceNumber buyer { name }'
});
```

(With the real fields we need in the `query`